src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1335,6 +1335,53 @@ bool ONNXMaxPoolSingleOutOp::inferShapes() {
 
 //===----------------------------------------------------------------------===//
 
+bool ONNXPadOp::inferShapes() {
+  // Cannot infer shape if no shape exists.
+  if (!data().getType().isa<RankedTensorType>()) {
+    emitError("Pad: unknown input shape");
+    return false;
+  }
+
+  // Cannot infer if the pads is not constant
+  DenseElementsAttr padsAttributes =
+      getAttr("pads").dyn_cast_or_null<mlir::DenseElementsAttr>();
+
+  if (!padsAttributes) {
```

Comment:
Such change requires all code which uses padAttributes to be moved into the then part. Since there are several error conditions, the level of nesting will be too deep, like this:

if (auto attr1 = ) {
   if (auto attr2 = ) {
    ....  //main part of code
   } else {
      error
   }
} else {
  error
}